server/drivers/hd44780-serial.c

```diff
@@ -362,16 +382,34 @@ serial_HD44780_scankeypad(PrivateData *p)
 					}
 					return retval;
 				}
-				else {
-					return buffer;
+				if (SERIAL_IF.connectiontype == HD44780_CT_EZIO) {
+					/* Support EZIO-100 0x4n and EZIO-300 0xBn */
+					switch (buffer) {
+						case 0x4B:
+						case 0xBB:
+							return 0x14;	/* KeyMatrix_4_1=Enter  */
+						case 0x4D:
+						case 0xBE:
+							return 0x24;	/* KeyMatrix_4_2=Up     */
+						case 0x47:
+						case 0xBD:
+							return 0x34;	/* KeyMatrix_4_3=Down   */
+						case 0x4E:
+						case 0xB7:
+							return 0x44;	/* KeyMAtrix_4_4=Escape */
+						/* No key 0x4F/0xBF or more than one key */ 
+						default:
+							return 0;
+					}
 				}
```

Comment:
Coding style want's the case statement on the same indentation level as the switch statement.